eth_account/hdaccount/mnemonic.py

```diff
@@ -0,0 +1,178 @@
+
+    return unicodedata.normalize("NFKD", utxt)
+
+
+class Mnemonic(object):
+    def __init__(self, language):
+        if language not in self.list_languages():
+            raise ValidationError(
+                f'Invalid language choice "{language}", must be one of {self.list_langauges()}'
+            )
+        self.language = language
+        self.radix = 2048
+        with open(WORDLIST_DIR / f"{language}.txt", "r", encoding="utf-8") as f:
+            self.wordlist = [w.strip() for w in f.readlines()]
+        if len(self.wordlist) != self.radix:
+            raise ValidationError(
+                f"Wordlist should contain {self.radix} words, "
+                f"but it contains {len(self.wordlist)} words."
+            )
+
+    @combomethod
+    def list_languages(_Mnemonic):
+        return sorted(Path(f).stem for f in WORDLIST_DIR.rglob("*.txt"))
+
+    @classmethod
+    def detect_language(cls, raw_mnemonic):
+        mnemonic = normalize_string(raw_mnemonic)
+        words = mnemonic.split(" ")
+        languages = cls.list_languages()
+
+        language_counts = {lang: 0 for lang in languages}
+        for lang in languages:
+            wordlist = cls(lang).wordlist
+            for word in words:
+                if word in wordlist:
+                    language_counts[lang] += 1
+
+        # No language had all words match it, so the language can't be fully determined
+        if max(language_counts.values()) < len(words):
+            raise ValidationError("Language not detected")
+
+        # Because certain wordlists share some similar words, we detect the language
+        # by seeing which of the languages have the most hits, and returning that one
+        most_matched_language = max(language_counts.items(), key=lambda c: c[1])[0]
+        return most_matched_language
+
+    def generate(self, num_words=12):
+        if num_words not in VALID_WORD_LENGTHS:
+            raise ValidationError(
+                f"Invalid choice for number of words: {num_words}, should be one of "
+                f"{VALID_WORD_LENGTHS}"
+            )
+        return self.to_mnemonic(os.urandom(4 * num_words // 3))  # 4/3 bytes per word
+
+    def to_mnemonic(self, seed):
+        if len(seed) not in VALID_SEED_SIZES:
+            raise ValidationError(
+                f"Invalid data length {len(seed)}, should be one of "
+                f"{VALID_WORD_LENGTHS}"
+            )
+        checksum = hashlib.sha256(seed).hexdigest()
+        bits = (
+            bin(int(binascii.hexlify(seed), 16))[2:].zfill(len(seed) * 8) +
+            bin(int(checksum, 16))[2:].zfill(256)[: len(seed) * 8 // 32]
+        )
+        result = []
+        for i in range(len(bits) // 11):
+            idx = int(bits[i * 11: (i + 1) * 11], 2)
+            result.append(self.wordlist[idx])
+        if self.language == "japanese":  # Japanese must be joined by ideographic space.
+            result_phrase = u"\u3000".join(result)
+        else:
```

Comment:
I actually don't think this would be the right place for it, because the language has already been validated in loading the class, and the `else` case is the correct behavior. In other words, there should never be a case where an exception needs to be raised.